src/matchers/tables.lua

```diff
@@ -1,13 +1,131 @@
+local prettyValue = require("src.utils.prettyValue")
+local deepEqual = require("src.utils.deepEqual")
+local assertType = require("src.asserts.type")
+
 ---@type lest.Matcher
-local function toHaveLength(ctx, received, length) end
+local function toHaveLength(ctx, received, length)
+	assertType(length, "number")
+
+	local success, comparison = pcall(function()
+		return #received == length
+	end)
+
+	return {
+		pass = success and comparison,
+		message = string.format(
+			"Expected %s to%shave a length of %d",
+			prettyValue(received),
+			ctx.inverted and " not " or " ",
+			length
+		),
+	}
+end
 
 ---@type lest.Matcher
-local function toContain(ctx, received, item) end
+local function toContain(ctx, received, item)
+	local pass = false
+
+	if type(received) == "string" then
+		assertType(item, "string")
```

Comment:
This is _indirectly_ asserting on the value of `received`

Simplest solution would be to use `tostring` on `item`, which may introduce some unexpected behaviour but will probably make more sense generally.

Reply:
what the fuck are you talking about
OK RESOLVED ANYWAYS

Reply:
We're changing what we assert based on the received value, meaning the matcher may throw a generic error depending on what we received.

This can be thought of similarly to using React hooks inside conditions.